back/engines/commercial/content_builder/app/models/content_builder/layout.rb

```diff
@@ -0,0 +1,26 @@
+# == Schema Information
+#
+# Table name: content_builder_layouts
+#
+#  id                     :uuid             not null, primary key
+#  craftjs_jsonmultiloc   :jsonb
+#  content_buildable_type :string           not null
+#  content_buildable_id   :uuid             not null
+#  code                   :string           not null
+#  enabled                :boolean          default(FALSE), not null
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#
+# Indexes
+#
+#  index_content_builder_layouts_content_buidable_type_id_code  (content_buildable_type,content_buildable_id,code) UNIQUE
+#
+module ContentBuilder
+  class Layout < ApplicationRecord
+    validates :content_buildable_type, :content_buildable_id, :code, presence: true
```

Comment:
Is there a reason to no model the relationship with projects? For polymorphic relationships we have the convention to still model the relationship and specify `polymorphic: true` (for example the voteable of a vote). The presence constraint could then be expressed on `content_buildable` instead of `content_buildable_type` and `content_buildable_id`. This allows us to detect layouts of a deleted project (in the case the layout attributes don't get cleared on project deletion).

I think it's also necessary to patch the project model to add the relationship there, so that `dependent: :destroy` can be specified, so that the layouts are deleted when their project is deleted.

Reply:
> Is there a reason to no model the relationship with projects? ...

No. Your feedback is very good. I changed the code.

Reply:
> I think it's also necessary to patch the project model to add the relationship there, so that dependent: :destroy can be specified, so that the layouts are deleted when their project is deleted.

Correct.

I can add
```
  has_many :content_builder_layouts,
            class_name: 'ContentBuilder::Layout',
            foreign_key: 'content_buildable_id',
            dependent: :destroy
```

but let's discuss how to test this properly.

Reply:
> I think it's also necessary to patch the project model to add the relationship there, so that dependent: :destroy can be specified, so that the layouts are deleted when their project is deleted.

Done.